internal/catalogmetadata/client/client.go

```diff
@@ -128,6 +159,10 @@ func PopulateExtraFields(catalogName string, channels []*catalogmetadata.Channel
 		}
 	}
 
+	for _, bundle := range bundles {
+		slices.SortFunc(bundle.InChannels, func(a, b *catalogmetadata.Channel) int { return cmp.Compare(a.Name, b.Name) })
+	}
```

Comment:
Could you elaborate on the purpose of this?

Reply:
Because of the concurrency introduced in `WalkMetasFS` the order that channels are added processed is non-deterministic. That non-determinism manifests as non-deterministic ordering of `bundle.InChannels`. So this function gets us back to a deterministic order.

It doesn't _really_ matter except that tests become more annoying to implement if the order isn't deterministic.

Reply:
Would you mind adding a comment just highlighting the why?

Reply:
https://github.com/operator-framework/operator-controller/compare/0661a996c07672f58b91a4e18ac35ba00710fd43..3df60bbed3d17645aec5eae832f884bb51a7b9a9